Sources/AsyncAlgorithms/Locking.swift

```diff
@@ -13,19 +13,23 @@
 import Darwin
 #elseif canImport(Glibc)
 import Glibc
+#elseif canImport(Musl)
+import Musl
 #elseif canImport(WinSDK)
 import WinSDK
+#else
+#error("Unsupported platform")
 #endif
 
 internal struct Lock {
 #if canImport(Darwin)
   typealias Primitive = os_unfair_lock
-#elseif canImport(Glibc)
+#elseif canImport(Glibc) || canImport(Musl)
   typealias Primitive = pthread_mutex_t
 #elseif canImport(WinSDK)
   typealias Primitive = SRWLOCK
 #else
-  typealias Primitive = Int
+  #error("Unsupported platform")
```

Comment:
From gitblame it seem like this was added for web assembly:
https://github.com/apple/swift-async-algorithms/pull/273/files
[Which seems that this can be checked for with ](https://github.com/swiftlang/swift/blob/34588acedb285cf02c6539db6fa1036d54237c6e/test/Parse/ConditionalCompilation/wasm32Target.swift#L4)arch(wasm32)[ or ](https://github.com/swiftlang/swift/blob/34588acedb285cf02c6539db6fa1036d54237c6e/test/Parse/ConditionalCompilation/wasm32Target.swift#L4)os(WASI)
[There is also an underscore check for multithreading](https://github.com/swiftlang/swift/blob/34588acedb285cf02c6539db6fa1036d54237c6e/test/Parse/ConditionalCompilation/wasm32ThreadsTarget.swift#L4): #if _runtime(_multithreaded)
This was added in https://github.com/swiftlang/swift/pull/72649

swift-testing uses a combination of os(WASI) and _runtime(_multithreaded) combined with compiler(>=6.1)
https://github.com/swiftlang/swift-testing/pull/322/files

FWIW Mutex has just landed in Swift 6.0

Reply:
cc: @0xpablo